pyquil/paulis.py

```diff
@@ -97,6 +97,22 @@ def __len__(self):
         """
         return len(self._ops)
 
+    def copy(self):
+        """
+        Properly creates a new PauliTerm, with a completely new dictionary
+        of operators
+        """
+        new_term = PauliTerm("I", 0, 0)
+        # coefficient and _id are copied as new refs (numbers, strings immutable)
+        new_term.coefficient = self.coefficient
+        new_term._id = self._id
+        # _ops dict needs to use own copy method
+        new_term._ops = self._ops.copy()
+
+        assert new_term == self  # check for value equality
+        assert not new_term._ops is self._ops  # check for ref inequality
```

Comment:
I don't think we need these assert statements here unless you had a specific problem you ran into? If so can you describe what the issue was?

The asserts can be cover with a unit test.